src/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java

```diff
@@ -72,6 +72,13 @@
     */
     private final static GenotypeLikelihoodsAllelePair[] PLIndexToAlleleIndex = calculatePLcache(MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED);
 
+
+    /*
+    * a cache of the PL index to the anyploid alleles it represents over all possible numbers of alternate alleles
+    */
+    protected final static List<List<Integer>> plIndexToAlleleIndices = new ArrayList<List<Integer>>();
```

Comment:
So you'll also store a static `ploidy` variable that will tell you for what ploidy the last is initialized and if it doesn't match, you'll reinitialize it? is that the idea? I'm not sure what the advantage is. Using a map will be thread safe will be just as "lazy" and allow for the odd case where two different ploidies are requested to remain as efficient. happy to be convinced otherwise...just interested in hearing the reasoning.